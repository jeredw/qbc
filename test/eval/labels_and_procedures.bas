10 print foo
20 function foo ()
if i = 1 then exit function
i = 1
print "hi"
goto 20
end function
