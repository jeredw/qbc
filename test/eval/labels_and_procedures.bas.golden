(program 
  (label 
    (line_number 10)) 
  (statement 
    (print_statement print 
      (expr 
        (variable_or_function_call foo)))) \n 
  (function_statement 
    (label 
      (line_number 20)) function foo 
    ( ) 
    (block \n 
      (statement 
        (if_inline_statement if 
          (expr 
            (expr 
              (variable_or_function_call i)) = 
            (expr 
              (literal 1))) then 
          (if_inline_action 
            (statement 
              (exit_statement exit function))))) \n 
      (statement 
        (assignment_statement 
          (variable_or_function_call i) = 
          (expr 
            (literal 1)))) \n 
      (statement 
        (print_statement print 
          (expr 
            (literal "hi")))) \n 
      (statement 
        (goto_statement goto 
          (target 
            (line_number 20)))) \n) 
    (end_function_statement end function)) \n <EOF>)
